src/components/icon-button/icon-button.tsx

```diff
@@ -0,0 +1,22 @@
+import clsx from 'clsx';
+
+import { Icon, IIconProps } from '../icon';
+
+ interface IconButtonProps extends IIconProps {
+  /**
+     * Optional send custom classes
+     */
+   classNames?: string,
+  /**
+   * Optional click handler
+   */
+  onClick?: () => void;
+}
+
+export const IconButton = ({ variant, size, classNames, onClick, }: IconButtonProps) => {
+  return (
+    <div className={clsx(classNames, "cursor-pointer rounded-m p-xs")} onClick={onClick}>
```

Comment:
statt `<div>` bitte `<button>` verwenden, wegen accesability.  
und `aria-label` verwenden

Reply:
- background fehlt
- :hover klassen fehlen
- :focus klassen fehlen
- :disabled klasse und prop fehlt

Reply:
statt `clsx` besser `cn` verwenden und `className` immer als letztes mitgeben, damit die klassen sauber gemerged werden können 
```ts
cn("cursor-pointer ...", className)

Reply:
Was meinst du mit focus  und background klassen? was sollen die machen?
Man kann in className senden oder?

Reply:
die komponente soll genau wie im figma erstellt werden: https://www.figma.com/file/nsXR2h0KwciWpuwKRD58FX/Mumble?type=design&node-id=413-288&mode=design&t=kA1UCI7Jse8yL2io-0  

die `className` prop soll nur optional sein und nur in spezalfällen verwendet werden.  
Die komponente soll alle zustände wie im figma abbilden können.  
somit soll hover state & focus state direkt in der componente als style mitgegeben werden. 

Schau dir gerne mal mein Beispiel an vom Button im feature/button unter `components/button-imporved` 

momentan fehlen noch einige sachen:
<img width="656" alt="image" src="https://github.com/ost-cas-fee-adv-23-24/design-system-component-library-team-batman/assets/73725192/89bd5c2e-3077-4cef-8a0d-35a5bff4defc">

so soll es aussehen:
<img width="231" alt="image" src="https://github.com/ost-cas-fee-adv-23-24/design-system-component-library-team-batman/assets/73725192/bd4c437e-bf14-4e2a-8542-7a6fa995942f">

- rounded-full -> müsste man noch in der tailwind config ergänzen (border radius 100%)
- icon padding `p-s`
- icon color `fill-white`
- `transition-color duration-300``
- `bg-base-600`
- `outline-offset-0`
- `hover:outline-[3px]` (spezialfall weil 3px nicht in spacing specs vorhanden)
- `hover:bg-base-700`
- `focus:outline-4`
- `focus:bg-base-700`